### PR TITLE
Clean up examples and add puppeteer tests

### LINKED ISSUE #5819: Jupyter Services Example appears broken
**Describe the bug**
The example found at https://github.com/jupyterlab/jupyterlab/tree/master/packages/services/examples/typescript-browser-with-output doesn't appear to work as advertised any longer.

**To Reproduce**
Follow the steps within https://github.com/jupyterlab/jupyterlab/blob/master/packages/services/examples/typescript-browser-with-output/README.md

**Expected behavior**
Expect the example to work with current version of jupyterlab packages

### PR DESCRIPTION
**Pull request by blink1073**
## References
Fixes #5819

## Code changes
Update example  tsconfig and webpack.   Cleans up all of the examples to be browser-testable and adds test for all 10 examples.

## User-facing changes
All of the examples now build and run without error :smile:

## Backwards-incompatible changes
NA

### CODE REVIEW

File: scripts/travis_script.sh [hunk 1]
```diff
@@ -26,6 +26,17 @@ if [[ $GROUP == js ]]; then
...
+    pushd packages/services/examples/typescript-browser-with-output
+    jlpm build
+    python main.py &
+    TASK_PID=$!
+    # Make sure the task is running
+    ps -p $TASK_PID || exit 1
+    sleep 5
+    kill $TASK_PID
+    wait $TASK_PID
+
```
Comment (ian-r-rose) [inline scripts/travis_script.sh]: Should we `popd` here? Doesn't matter much right now since it's at the end, but could in the future.

**Comment by ian-r-rose** [comment]
I took a pass at the other services tests and ran into a couple of issues:
1. The `browser` example currently fails because `tsc` is missing from its dependencies in the `package.json`. It might pass locally if the user has a reasonably up-to-date `tsc` installed globally.
2. I couldn't get the `node` example working because of an `_xsrf` error. I'm not sure why this is happening, since the token is an empty string in this example. Possibly related to #6141?

**Comment by blink1073** [comment]
Thanks for the testing!  I'll dig into those other ones as well.

**Comment by blink1073** [comment]
@ian-r-rose, the xrsf issue arose if you ran that script from a different directory, now fixed.

I went ahead and added automated testing for all 10 examples :smile:

File: examples/example_check.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
+
```
Comment (vidartf) [inline examples/example_check.py]: Would you mind adding a small docstring explaining what the purpose of this file is? It makes it a lot easier to navigate the helper files!

File: examples/test_examples.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
+
```
Comment (vidartf) [inline examples/test_examples.py]: Same here, a docstring would be helpful.

**Comment by blink1073** [comment]
Good call, docstrings added.

**Comment by vidartf** [comment]
LGTM!

File: packages/services/examples/typescript-browser-with-output/src/index.ts [hunk 4]
```diff
@@ -16,25 +17,20 @@ import {
...
 
-  void Kernel.startNew().then(kernel => {
-    outputArea.future = kernel.requestExecute({ code });
-    document.getElementById('outputarea').appendChild(outputArea.node);
-  });
+  const kernel = await Kernel.startNew();
+  outputArea.future = kernel.requestExecute({ code });
+  document.getElementById('outputarea').appendChild(outputArea.node);
+  await outputArea.future.done;
+  console.log('Test complete!');
```
Comment (SimonBiggs) [inline packages/services/examples/typescript-browser-with-output/src/index.ts]: Thanks for this @blink1073 :). Really helps me jump back into programming with @jupyterlab tools with the new changes :).

Thank you!

Comment (blink1073) [reply packages/services/examples/typescript-browser-with-output/src/index.ts]: No problem @SimonBiggs, nice to e-see you again!

Comment (SimonBiggs) [reply packages/services/examples/typescript-browser-with-output/src/index.ts]: :). Also, this does remind me, I do have quite a bit else to thank you for. I learnt so much off of all the work you've done here. For example I have used what I learnt about Lerna here to implement a Python monorepo https://github.com/pymedphys/pymedphys/tree/master/packages. I couldn't have done it without your example, so thank you :).

I also got the ideas of dependency graphing off of you guys:

https://pymedphys.com/developer/dependencies.html

And many, many other things. So, thank you. :) Thanks @blink1073 :).

Comment (blink1073) [reply packages/services/examples/typescript-browser-with-output/src/index.ts]: My pleasure.  One day we'll meet and you can buy me a beer.  ;)

Comment (SimonBiggs) [reply packages/services/examples/typescript-browser-with-output/src/index.ts]: Now that, is a good plan :)

**Pull request merged**